src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -151,6 +151,17 @@ private void OnStatusChanged(bool started)
         return await connection.TryInvokeAsync(project, invocation, cancellationToken).ConfigureAwait(false);
     }
 
+    public async ValueTask<Optional<TResult>> TryInvokeAsync<TService, TResult>(
+        SolutionCompilationState compilationState,
+        ProjectId projectId,
+        Func<TService, Checksum, CancellationToken, ValueTask<TResult>> invocation,
+        CancellationToken cancellationToken)
+        where TService : class
+    {
+        using var connection = CreateConnection<TService>(callbackTarget: null);
+        return await connection.TryInvokeAsync(compilationState, projectId, invocation, cancellationToken).ConfigureAwait(false);
+    }
```

Comment:
helper overload for the case where we're asking about a particular project in the context of a particular SolutionCompilationState (first time we've needed to do that).